.github/workflows/integration-limited-juju3.yaml

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+on:
+  workflow_call:
+    inputs:
+      juju-snap-channel:
+        description: Snap channel for Juju CLI
+        default: 3.1/stable
+        type: string
+      libjuju-version-specifier:
+        description: |
+            python-libjuju version specifier (e.g. ">=1.3")
+            https://packaging.python.org/en/latest/glossary/#term-Version-Specifier
+        required: false
+        type: string
+
+jobs:
+  build:
+    name: Build charms
+    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v2
+
+  integration-test:
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        tox-environments:
+          - charm-integration
+          - tls-integration
+          - client-integration
+        runner: ["ubuntu-22.04"]
```

Comment:
As discussed:
Test execution limited to items that relate to secrets (i.e. the core feature we're using from Juju 3 yet) in order to offload pipelines.